### PR TITLE
Feature/add get documents api

### PR DESCRIPTION
**Pull request by Dhruvish0146**

### CODE REVIEW

File: src/document/document.controller.ts [hunk 1]
```diff
@@ -23,6 +26,17 @@ const storage = multer.diskStorage({
   },
 });
 
+const convertStringToInt = (data: string): number => {
+  let parsedData: number | undefined;
```
Comment (sandeepmehta2155) [inline src/document/document.controller.ts:30]: Number(parsedData)

File: src/document/document.controller.ts [hunk 2]
```diff
@@ -15,6 +15,9 @@ import { FileInterceptor } from '@nestjs/platform-express';
 import { Prisma, document } from '@prisma/client';
 import { DocumentService } from './document.service';
 import { CreateDocumentDto } from './dto/create-document.dto';
+import { ApiQuery } from '@nestjs/swagger';
```
Comment (sandeepmehta2155) [inline src/document/document.controller.ts:18]: Fix this import sequence

File: src/document/document.controller.ts [hunk 3]
```diff
@@ -59,46 +73,67 @@ export class DocumentController {
...
+      department,
+      start_date,
+      end_date,
+      start_year,
+      end_year,
+      start_month,
+      end_month,
+      sort_by,
+    } = query;
+    let parsedPage =
```
[the remark this answers is not in the document]
Comment (sandeepmehta2155) [reply src/document/document.controller.ts:95]: use interceptor

File: src/document/dto/get-user-documents.dto.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { IsNumber, Max, Min } from 'class-validator';
+
+export enum status_type {
```
Comment (sandeepmehta2155) [inline src/document/dto/get-user-documents.dto.ts:4]: no enums in dto

File: src/document/dto/get-user-documents.dto.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  PENDING,
+  APPROVE,
+  REJECT,
+}
+export class GetUserDocuemts {
+  @ApiPropertyOptional({
+    type: Number,
+  })
```
Comment (sandeepmehta2155) [inline src/document/dto/get-user-documents.dto.ts:12]: Add required validators

File: src/document/document.controller.ts [hunk 6]
```diff
@@ -59,46 +73,67 @@ export class DocumentController {
     }
   }
 
-  @Get(':id')
-  async getDocuments(
+  @Get('user/:id')
```
Comment (sandeepmehta2155) [inline src/document/document.controller.ts:76]: Add request/response dto

File: src/document/document.controller.ts [hunk 7]
```diff
@@ -9,12 +9,14 @@ import {
...
   UseInterceptors,
 } from '@nestjs/common';
-import * as multer from 'multer';
-import * as fs from 'fs';
 import { FileInterceptor } from '@nestjs/platform-express';
-import { Prisma, document } from '@prisma/client';
 import { DocumentService } from './document.service';
 import { CreateDocumentDto } from './dto/create-document.dto';
+import { ApiResponse } from '@nestjs/swagger';
+import { GetUserDocuemts } from 'src/document/dto/get-user-documents.dto';
```
Comment (Dhruvish0146) [inline src/document/document.controller.ts:16]: import sequence

File: src/document/dto/get-user-document-response.dto.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+  @IsDateString()
+  @IsNotEmpty()
+  @ApiProperty({ description: 'Document creation date and time', format: 'date-time' })
+  created_at: Date;
+
+  @IsString()
+  @IsOptional()
+  @ApiProperty({ description: 'Description of the document (optional)' })
+  discription: string;  // Typo correction
```
Comment (Dhruvish0146) [inline src/document/dto/get-user-document-response.dto.ts:28]: description spell check

**Pull request merged**